libsolidity/lsp/Transport.h

```diff
@@ -119,27 +143,34 @@ class IOStreamTransport: public Transport
 	/// @param _out for example std::cout (stdout)
 	IOStreamTransport(std::istream& _in, std::ostream& _out);
 
-	// Constructs a JSON transport using standard I/O streams.
-	IOStreamTransport();
-
 	bool closed() const noexcept override;
-	std::optional<Json::Value> receive() override;
-	void notify(std::string _method, Json::Value _params) override;
-	void reply(MessageID _id, Json::Value _result) override;
-	void error(MessageID _id, ErrorCode _code, std::string _message) override;
 
 protected:
-	/// Sends an arbitrary raw message to the client.
-	///
-	/// Used by the notify/reply/error function family.
-	virtual void send(Json::Value _message, MessageID _id = Json::nullValue);
-
-	/// Parses header section from the client including message-delimiting empty line.
-	std::optional<std::map<std::string, std::string>> parseHeaders();
+	std::string readBytes(size_t _byteCount) override;
+	std::string getline() override;
+	void writeBytes(std::string_view _data) override;
+	void flushOutput() override;
 
 private:
 	std::istream& m_input;
 	std::ostream& m_output;
 };
 
+/**
+ * Standard I/O transport Layer utilizing stdin/stdout for communication.
```

Comment:
The changes in these files do not look like they are about allowing full filesystem access. Please put them in their own PR in the future.